### PR TITLE
Minor dev experience improvement for Linux/Mac devs

### PR DESCRIPTION
**Pull request by jabba2324**
Some Linux distros (namely ubuntu) now only ship with python 3 using the command 'python3' this will cause your launch-game.sh script to fail at detecting the game engine path.

The following changes add a retry for python3 and some logging around when the path is empty.

### CODE REVIEW

File: OpenRA.Game/Game.cs [hunk 1]
```diff
@@ -360,6 +360,11 @@ static void Initialize(Arguments args)
 			{
 				var launchPath = args.GetValue("Engine.LaunchPath", Assembly.GetEntryAssembly().Location);
 
+				if(String.IsNullOrEmpty(launchPath))
+				{
+					Console.WriteLine("No launch path was specified, please check launch scripts.");
```
Comment (abcdefg30) [inline OpenRA.Game/Game.cs]: Our windows scripts don't set a launch path at all, so this warning would occur there all the time.

File: launch-game.sh [hunk 2]
```diff
@@ -1,6 +1,12 @@
 #!/bin/sh
 MODLAUNCHER=$(python -c "import os; print(os.path.realpath('$0'))")
 
+if [ -z "$MODLAUNCHER" ]
+then
+      echo "\$MODLAUNCHER is empty trying python3"
```
Comment (abcdefg30) [inline launch-game.sh]: Style nit: This should use tabs instead of spaces.

**Review by abcdefg30 — COMMENTED** [review]
I wonder if we should use python3 by default here, and either just remove the normal python way or use that as fallback.

**Comment by pchote** [comment]
Yes, python 2 is now obsolete so should not be treated as the go-to option.
I would prefer it if we could use our standard pattern for fallback commands:
```bash
if command -v python3 >/dev/null 2>&1; then
	 MODLAUNCHER=$(python3 -c "import os; print(os.path.realpath('$0'))")
else
	 MODLAUNCHER=$(python2 -c "import os; print(os.path.realpath('$0'))")
fi
```

It is more verbose, but makes the behaviour clear and avoids unnecessary warnings.

**Comment by Mailaender** [comment]
I can confirm that this Debian specific workaround will have no bad influence on @openSUSE. Both `python` and `python3` point to Python 3 by default.

File: OpenRA.Game/Game.cs (hunk 1 above)
Comment (jabba2324) [reply OpenRA.Game/Game.cs]: Do you know what the default value of launchPath is on Windows? If it was null or empty then the proceeding line 

`if (launchPath.First() == '"' && launchPath.Last() == '"')` 

Would always throw an exception as it currently does in Debian/Ubuntu? 

EDIT:
After a peek into the arguments class it seems that this would work fine on Windows as the default is 
`Assembly.GetEntryAssembly().Location `
which is implemented on Windows.

Comment (pchote) [reply OpenRA.Game/Game.cs]: My preference would be to not change this code at all - the issue is in the launch scripts.

**Comment by jabba2324** [comment]
@pchote I'll implement the approach given in your comment and remove the logging from Game.cs

**Comment by jabba2324** [comment]
I haven't forgotten about this, just waiting for a new PC to arrive as my old one bit the dust.

**Comment by pchote** [comment]
Can you please squash the commits together?

**Comment by jabba2324** [comment]
> Can you please squash the commits together?

Squashed

**Comment by pchote** [comment]
Sorry for the delay here. The commit message no longer reflects the contents of what it actually does, so can we reword this to something like "Add python3 support to launch-game.sh"?

This otherwise LGTM - tested on Ubuntu 20.04 and macOS 10.15.

**Comment by jabba2324** [comment]
> Sorry for the delay here. The commit message no longer reflects the contents of what it actually does, so can we reword this to something like "Add python3 support to launch-game.sh"?
> 
> This otherwise LGTM - tested on Ubuntu 20.04 and macOS 10.15.

I've opened a new PR here with an appropriately named branch and commit -m

https://github.com/OpenRA/OpenRA/pull/18375
